client/lib/signup/flow-controller.ts

```diff
@@ -442,7 +444,15 @@ export default class SignupFlowController {
 
 	_getNeedsToGoThroughCheckout() {
 		const progress = getSignupDependencyProgress( this._reduxStore.getState() );
-		return !! progress?.plans?.cartItem;
+
+		if ( progress?.plans ) {
+			return (
+				!! progress.plans?.cartItem ||
+				progress.plans?.cartItems?.find( ( item: MinimalRequestCartProduct ) => isPlan( item ) )
+			);
+		}
```

Comment:
I'm wondering if we really have to check whether there is a plan in `cartItems`. Shouldn't we always need to go through checkout if there is anything in the cart? 🤔

Reply:
Ah this was part of the cleanup I wanted to do today but couldn't get to yesterday. The answer to your question is I'm unsure, so I was going to review the code and verify through some testing 👍